source/openapi-admin-v3.yaml

```diff
@@ -5132,55 +5178,517 @@ components:
         - name
         - function_source
         - respond_result
-    UserProvider:
-      type: object
-      properties:
-        id:
-          type: string
-        provider_type:
-          "$ref": "#/components/schemas/ProviderType"
-        provider_id:
-          type: string
-    ProviderSummary:
+    
+
+
+
+
+
+
+
+
+
+    BaseAuthProvider:
+      required:
+        - name
+        - type
       properties:
         _id:
           type: string
+          description: The authentication provider's unique ID.
         name:
           type: string
           description: The name of the authentication provider. This value must be the same as the value of `type`.
         type:
           "$ref": "#/components/schemas/ProviderType"
         disabled:
           type: boolean
-    FullProvider:
+          description: If `true`, the auth provider is disabled. Users cannot log in through the provider until it is enabled.
+    
+    AuthProvider:
+      anyOf:
+        - $ref: "#/components/schemas/AnonymousAuthProvider"
+        - $ref: "#/components/schemas/EmailPasswordAuthProvider"
+        - $ref: "#/components/schemas/ApiKeyAuthProvider"
+        - $ref: "#/components/schemas/CustomJwtAuthProvider"
+        - $ref: "#/components/schemas/CustomFunctionAuthProvider"
+        - $ref: "#/components/schemas/AppleAuthProvider"
+        - $ref: "#/components/schemas/GoogleAuthProvider"
+        - $ref: "#/components/schemas/FacebookAuthProvider"
```

Comment:
is there anyway to do inheritance in the object oriented sense here?

would be elegant to be able say for example `FacebookAuthProvider extends AuthProvider`, and then ref `AuthProvider` in the endpoints. 

fun idea, i assume not possible, but just wanted to put it out there in case it is.